src/tools/pkcs11-tool.c

```diff
@@ -2651,7 +2651,10 @@ derive_key(CK_SLOT_ID slot, CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key)
 	CK_BBOOL false = FALSE;
 	CK_OBJECT_HANDLE newkey = 0;
 	CK_RV rv;
+	CK_ECDH1_DERIVE_PARAMS ecdh_parms;
 	int fd, r;
+	unsigned char* buf = NULL;  /* buffer for handling input data */
```

Comment:
The "buf" buffer is indirectly referred after the CKM_ECDH1_DERIVE case falls out of scope. That's why it has to be declared outside of the case. Look at the code just before p11->C_DeriveKey.
I made it dynamic thinking that if more mechanisms are added later, each can calculate and allocate the space they need depending on the algorithm and key length. 

See commit 2ba87640d23bcff3f1a130ea28bf02f4f8afbb3e By mouse07410, it fixes the same thing.